File: 900/1696B/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int cnt = 0;
            int curr = 0;
            
            for(int i=0;i<n;i++)
            {
                if(sc.nextInt() == 0)
                {
                    if(curr > 0)
                        cnt++;
                    curr = 0;
                }
                else
                    curr++;
            }

            if(curr > 0)
                cnt++;
                
            System.out.println(Math.min(2,cnt));
        }
    }
}

File: 900/1696B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;

        int cnt = 0;
        int curr = 0;

        for (int i = 0; i < n; i++) 
        {
            int x;
            cin >> x;
            if (x == 0) 
            {
                if (curr > 0) 
                    cnt++;
                
                curr = 0;
            } else 
                curr++;
        }

        if (curr > 0) 
            cnt++;

        cout << min(2, cnt) << "\n";
    }

    return 0;
}
